### PR TITLE
VEN-756 | Dependency upgrades and refactoring

### PR DESCRIPTION
**Pull request by TatuArvela**
## Description :sparkles:

* Upgrade several dependencies that could easily be upgraded
* Sort package.json
* Rename all `*.spec.*` files to `*.test.*`
* Replace TSLint with ESLint
* Remove serviceWorker.ts

### Closes :no_good_woman:

* [VEN-756](https://helsinkisolutionoffice.atlassian.net/browse/VEN-756): Upgrade more outdated dependencies

### CODE REVIEW

File: src/utils/unmarkedWinterStorage.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import { List } from 'immutable';
+import _ from 'lodash';
```
Comment (Nizar-Rahme) [inline src/utils/unmarkedWinterStorage.ts:2]: For a smaller bundle size:
```suggestion
import orderBy from 'lodash/orderBy';
```

File: src/components/forms/fragments/overview/bigShipsInfo/BigShipsInfo.tsx [hunk 2]
```diff
@@ -38,9 +31,7 @@ const BigShipsInfo = ({
       break;
   }
 
-  const rentingPeriodWDates = `${rentingPeriodLabel} ${rentFrom ? rentFrom : ''}-${
-    rentTill ? rentTill : ''
-  }`;
+  const rentingPeriodWDates = `${rentingPeriodLabel} ${rentFrom ? rentFrom : ''}-${rentTill ? rentTill : ''}`;
```
Comment (Nizar-Rahme) [inline src/components/forms/fragments/overview/bigShipsInfo/BigShipsInfo.tsx:34]:
```suggestion
  const rentingPeriodWDates = `${rentingPeriodLabel} ${rentFrom || ''}-${rentTill || ''}`;
```

File: src/components/forms/fragments/exchangeApplication/ExchangeApplicationContainer.tsx [hunk 3]
```diff
@@ -17,8 +17,7 @@ const ExchangeApplicationContainer = (props: Props) => {
   return (
     <Query<BerthSwitchReasonsQuery> query={BERTH_SWITCH_REASONS_QUERY}>
       {({ data }) => {
-        const reasons =
-          data && data.berthSwitchReasons ? data.berthSwitchReasons.filter(isReason) : [];
+        const reasons = data && data.berthSwitchReasons ? data.berthSwitchReasons.filter(isReason) : [];
```
Comment (Nizar-Rahme) [inline src/components/forms/fragments/exchangeApplication/ExchangeApplicationContainer.tsx:20]:
```suggestion
        const reasons = data?.berthSwitchReasons ? data.berthSwitchReasons.filter(isReason) : [];
```

**Review by Nizar-Rahme — CHANGES_REQUESTED** [review]
I left a couple of nitpicks, fantastic job overall 💎

File: src/@types/piwik-react-router.d.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ declare module 'piwik-react-router' {
     enableLinkTracking?: boolean;
     updateDocumentTitle?: boolean;
     ignoreInitialVisit?: boolean;
+    // eslint-disable-next-line @typescript-eslint/ban-types
```
[the remark this answers is not in the document]
Comment (TatuArvela) [reply src/@types/piwik-react-router.d.ts:8]: Yep, I'll remove this.
I actually temporarily disabled this rule globally, we need to fix each of those separately later.

**Review by Nizar-Rahme — APPROVED** [review]
Great improvements! 🎖️

**Pull request merged**